packages/jellyfish-api-core/__tests__/category/account.test.ts

```diff
@@ -0,0 +1,487 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../container_adapter_client'
+import waitForExpect from 'wait-for-expect'
+import BigNumber from 'bignumber.js'
+
+describe('masternode', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+    await container.waitForWalletBalanceGTE(300)
+
+    await setup()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  async function setup (): Promise<void> {
+    await utxosToAccount(50)
+    await utxosToAccount(60)
+    await utxosToAccount(190)
+
+    const from = await container.call('getnewaddress')
+    await createToken(from, 'DBTC', 200)
+
+    const to = await accountToAccount('DBTC', 5, from)
+    await accountToAccount('DBTC', 18, from, to)
+
+    await createToken(from, 'DETH', 200)
+    await accountToAccount('DETH', 46, from)
+  }
+
+  async function utxosToAccount (amount: number): Promise<void> {
+    const address = await container.call('getnewaddress')
+
+    const payload: { [key: string]: string } = {}
+    payload[address] = `${amount.toString()}@0`
+    await container.generate(25)
+  }
+
+  async function createToken (address: string, symbol: string, amount: number): Promise<void> {
+    const metadata = {
+      symbol,
+      name: symbol,
+      isDAT: true,
+      mintable: true,
+      tradeable: true,
+      collateralAddress: address
+    }
+    await container.call('createtoken', [metadata])
+    await container.generate(25)
+
+    await container.call('minttokens', [`${amount.toString()}@${symbol}`])
+    await container.generate(25)
+  }
+
+  async function accountToAccount (symbol: string, amount: number, from: string, _to = ''): Promise<string> {
+    const to = _to !== '' ? _to : await container.call('getnewaddress')
+
+    await container.call('accounttoaccount', [from, { [to]: `${amount.toString()}@${symbol}` }])
+    await container.generate(25)
+
+    return to
+  }
+
+  describe('listAccounts', () => {
+    it('should listAccounts', async () => {
+      await waitForExpect(async () => {
+        const accounts = await client.account.listAccounts({}, true, { indexedAmounts: false, isMineOnly: false })
+        expect(accounts.length).toBeGreaterThan(0)
+      })
+
+      const accounts = await client.account.listAccounts({}, true, { indexedAmounts: false, isMineOnly: false })
```

Comment:
and for this type of test, you should test without setting the default options. or else you miss test coverage for this.

https://app.codecov.io/gh/DeFiCh/jellyfish/compare/162/diff#diff-cGFja2FnZXMvamVsbHlmaXNoLWFwaS1jb3JlL3NyYy9jYXRlZ29yeS9hY2NvdW50LnRz

This applyies for all the default cases.

Reply:
phew.. i guess setting the method with optional paras on first order is a rule, its more flexible